allennlp/common/params.py

```diff
@@ -351,13 +351,14 @@ def _check_is_dict(self, new_history, value):
         return value
 
     @staticmethod
-    def from_file(params_file: str, params_overrides: str = "") -> 'Params':
+    def from_file(params_file: str, params_overrides: str = "", ext_vars: dict = {}) -> 'Params':
+        # pylint: disable=dangerous-default-value
         """
         Load a `Params` object from a configuration file.
         """
         # redirect to cache, if necessary
         params_file = cached_path(params_file)
-        ext_vars = dict(os.environ)
+        ext_vars = {**dict(os.environ), **ext_vars}
```

Comment:
This breaks the API for passing environment variables _not_ via beaker.

Reply:
how does it break it? it's saying "get all the vars from the environment, then override them with `ext_vars` if provided"

Reply:
How does this break the API?

Reply:
Oh sorry I misread the diff. Nevermind.